Library/Homebrew/formula.rb

```diff
@@ -657,6 +663,29 @@ def skip_clean?(path)
     self.class.skip_clean_paths.include? to_check
   end
 
+  # Sometimes we accidentally install files outside prefix. After we fix that,
+  # users will get nasty link conflict error. So we create a whitelist here to
+  # allow overwriteing certain files. e.g.
+  #   link_overwrite "bin/foo", "lib/bar"
+  #   link_overwrite "share/man/man1/baz-*"
+  def link_overwrite?(path)
+    # Don't overwrite files for a fresh installation.
+    return false unless any_old_version_installed?
```

Comment:
I think we should definitely replace them and I don't think we really need backup logic either. If we do insist on adding the backup logic they should be put in another directory rather than just have an extension appended and be cleaned up by `brew cleanup`. Making the users manually delete them isn't nice (and may actually break some software that iterates over the contents of a directory for e.g. plugins).